x/nexus/keeper/rate_limit.go

```diff
@@ -16,7 +16,7 @@ import (
 )
 
 // RateLimitTransfer applies a rate limit to transfers, and returns an error if the rate limit is exceeded
-func (k Keeper) RateLimitTransfer(ctx sdk.Context, chain exported.ChainName, asset sdk.Coin, direction exported.TransferDirection) error {
+func (k Keeper) RateLimitTransfer(ctx sdk.Context, chain exported.ChainName, asset sdk.Coin, direction exported.TransferDirection, messageId string) error {
```

Comment:
I don't like having an additional parameter that looks mandatory but isn't and is only used for logging

Reply:
Agreed it's not ideal. I would like to log this since finding the tx has been confusing from the old log. I can make it a default argument (but I find go's default convention more confusing) and we want to make the caller pass a value if it has one and not forget about it. Suggestions?